pennylane/circuit_graph.py

```diff
@@ -469,9 +471,42 @@ def get_depth(self):
                 self._operation_graph = self._graph.subgraph(
                     list(self._graph.nodes().index(node) for node in self.operations)
                 )
+                self._extend_graph(self._operation_graph)
                 self._depth = rx.dag_longest_path_length(self._operation_graph) + 1
         return self._depth
 
+    def _extend_graph(self, graph: rx.PyDiGraph) -> rx.PyDiGraph:
+        """Extend graph to account for custom depth operations"""
+        custom_depth_node_dict = dict()
+        for op in self.operations:
+            if isinstance(op, OperationWithResources) and (d := op.resources().depth) > 1:
+                custom_depth_node_dict[graph.nodes().index(op)] = d
```

Comment:
I had questions about performance concerns for users outside of resource estimation workflows, but in that case `custom_depth_node_dict` would be empty, right?

Reply:
Correct! As long as there aren't any `OperationsWithResources` on the tape, the dict will remain empty and the circuit graph will be the same 👍🏼